### PR TITLE
CEDS-2246 Make Declarant Details optional and remove non existing address

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: app/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilder.scala [hunk 1]
```diff
@@ -36,52 +35,16 @@ class DeclarantBuilder @Inject()(countriesService: CountriesService) extends Mod
...
-
-  private def deriveCountryCode(addressCountry: String): String =
-    countriesService.allCountries
-      .find(country => addressCountry.contains(country.countryName))
-      .map(_.countryCode)
-      .getOrElse("")
-
-  private def isDefined(declarantDetails: DeclarantDetails): Boolean =
-    declarantDetails.details.eori.isDefined || declarantDetails.details.address.isDefined
+  private def isDefined(declarantDetails: DeclarantDetails): Boolean = declarantDetails.details.flatMap(_.eori).isDefined
```
Comment (MaciejRewera) [inline app/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilder.scala]: Is it used anywhere now?

File: app/uk/gov/hmrc/exports/models/declaration/Parties.scala [hunk 2]
```diff
@@ -19,31 +19,37 @@ package uk.gov.hmrc.exports.models.declaration
...
 }
 
 case class ConsigneeDetails(details: EntityDetails)
+
 object ConsigneeDetails {
   implicit val format: OFormat[ConsigneeDetails] = Json.format[ConsigneeDetails]
 }
 
-case class DeclarantDetails(details: EntityDetails)
+case class DeclarantDetails(details: Option[EntityDetails])
```
Comment (MaciejRewera) [inline app/uk/gov/hmrc/exports/models/declaration/Parties.scala]: Do we have a scenario when we have no declarant details? I thought this is mandatory for the whole declaration, no matter what type it has. How would we be able to tell who has submitted it?

File: app/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilder.scala (hunk 1 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilder.scala]: Yes, in the first method in this class

File: app/uk/gov/hmrc/exports/models/declaration/Parties.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/models/declaration/Parties.scala]: DeclarantDetails are optional in Parties. This element is mandatory in every journey except Clearance

Comment (MaciejRewera) [reply app/uk/gov/hmrc/exports/models/declaration/Parties.scala]: Besides, in the `ExportsDeclaration` model this field is already optional, so why would we need to have so many nested `Option`s ?

File: test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala [hunk 3]
```diff
@@ -16,75 +16,57 @@
...
-        declaration.getDeclarant should be(null)
+        declaration.getDeclarant shouldBe null
       }
 
-      "no eori" in {
-        val model = aDeclaration(withDeclarantDetails(eori = None, address = Some(Address("name", "line", "city", "postcode", "United Kingdom"))))
-        val declaration = new Declaration()
-
-        builder.buildThenAdd(model, declaration)
+      "empty declarant details" in {
```
Comment (MaciejRewera) [inline test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala]: This test case and the next one are exactly the same.

File: test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala [hunk 4]
```diff
@@ -16,75 +16,57 @@
...
         val declaration = new Declaration()
 
         builder.buildThenAdd(model, declaration)
 
-        declaration.getDeclarant.getID should be(null)
-        declaration.getDeclarant.getAddress.getCountryCode.getValue should be("")
+        declaration.getDeclarant shouldBe null
       }
 
       "populated" in {
```
Comment (MaciejRewera) [inline test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala]: Nit: What do you think about changing this bit? To me, "populated" is more suitable for views.

File: test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala (hunk 3 above)
Comment (PatrykRudnicki) [reply test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala]: empty declarant details means empty but visited. The model is different. Behaviour the same

File: test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala (hunk 4 above)
Comment (PatrykRudnicki) [reply test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala]: Hmm, not sure. Any proposals?

File: app/uk/gov/hmrc/exports/models/declaration/Parties.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/models/declaration/Parties.scala]: That's a good question. As I understand this approach allow us to present empty element on the summary page. The same approach is on the representative details

**Comment by PatrykRudnicki** [comment]
Closing for now, probably unnecessary change

File: test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala (hunk 3 above)
Comment (MaciejRewera) [reply test/unit/uk/gov/hmrc/exports/services/mapping/declaration/DeclarantBuilderSpec.scala]: Ok I see

**Pull request closed without merging**